### PR TITLE
BREAKING CHANGE: migrate to fastify v5

### PR DESCRIPTION
**Pull request by ilteoood**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -86,9 +86,6 @@ function fastifyOverview (fastify, options, next) {
     instance[kTrackerMe] = trackingToken
 
     const trackStructure = getPluginNode(trackingToken, instance.pluginName)
-    if (opts.addSource && this) {
```
Comment (Eomm) [inline index.js:89]: Could you update the README?

(I would add a `note` that the `addSource` flag has been removed in fastify v5

Comment (ilteoood) [reply index.js:89]: AddSource is actually used elsewhere

Comment (ilteoood) [reply index.js:89]: @Eomm an example is on line 35 of this file

File: test/expose-routes.test.js [hunk 2]
```diff
@@ -9,106 +9,117 @@ test('expose the route to the the json', async t => {
...
-          onError: [],
-          onSend: [],
-          onResponse: [],
-          onTimeout: []
-        }
+  const responseContent = response.json()
+
+  t.assert.deepStrictEqual(responseContent.name, 'fastify -> fastify-overview')
+  t.assert.deepStrictEqual(responseContent.children, [])
+  t.assert.deepEqual(responseContent.routes, [
```
Comment (Eomm) [inline test/expose-routes.test.js:16]: the `hasStrinct` is like `partialDeepStrictEqual` (in nodejs 23).
I would add a `TODO` comment and just:

- verify that the `preHandler` has 1 item
- remove the item
- keep the deepEqual as it is now

File: test/expose-routes.test.js [hunk 3]
```diff
@@ -9,106 +9,117 @@ test('expose the route to the the json', async t => {
...
+        onError: [],
+        onRequest: [],
+        onRequestAbort: [],
+        onResponse: [],
+        onSend: [],
+        onTimeout: [],
+        preHandler: [{
+          name: 'hook',
+          hash: '1994f633554c479b3ffceeebdb7328ddf56d0234'
+        }],
```
Comment (Eomm) [inline test/expose-routes.test.js:95]:
```suggestion
        preHandler: [],
```

**Pull request merged**